src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,181 @@
+-- overlayD (1 * 2 + 1 * 3) (1 * 2) == (Left 1) * (Left 2) 
+--                                   + (Left 1) * (Left 3)
+--                                   + (Right 1) * (Right 2)
+-- @
+overlayD ::
+     (Ord a, Ord b)
+  => AdjacencyMap a
+  -> AdjacencyMap b
+  -> AdjacencyMap (Either a b)
+overlayD (AAM a) (AAM b) = AAM (AM.overlay (AM.gmap Left a) (AM.gmap Right b))
+
+-- | Perform a disjoint connect of two different acyclic graphs.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- connectD empty empty     == 'empty'
+-- connectD (1 + 2) (1 + 2) == (Left 1) * (Right 1) 
+--                           + (Left 1) * (Right 2)
+--                           + (Left 2) * (Right 1)
+--                           + (Left 2) * (Right 2)
+-- connectD (1 * 2) (1 * 2) == (Left 1) * (Right 1)
+--                           + (Left 1) * (Right 2) 
+--                           + (Left 2) * (Right 1) 
+--                           + (Left 2) * (Right 2) 
+--                           + (Left 1) * (Left 2) 
+--                           + (Right 1) * (Right 2)
+-- @
+connectD ::
+     (Ord a, Ord b)
+  => AdjacencyMap a
+  -> AdjacencyMap b
+  -> AdjacencyMap (Either a b)
+connectD (AAM a) (AAM b) = AAM (AM.connect (AM.gmap Left a) (AM.gmap Right b))
+
+-- | Compute the /transitive closure/ of a graph.
+-- Complexity: /O(n * m * log(n)^2)/ time.
+--
+-- @
+-- transitiveClosure 'empty'             == 'empty'
+-- transitiveClosure ('vertex' x)        == 'vertex' x
+-- transitiveClosure (1 * 2 + 2 * 3)     == 1 * 2 + 2 * 3 + 1 * 3
+-- transitiveClosure . transitiveClosure == transitiveClosure
+-- @
+transitiveClosure :: Ord a => AdjacencyMap a -> AdjacencyMap a
+transitiveClosure (AAM x) = AAM (AM.transitiveClosure x)
+
+-- | Compute the /condensation/ of a graph, where each vertex
+-- corresponds to a /strongly-connected component/ of the original
+-- graph. Note that component graphs are non-empty, and are therefore
+-- of type "Algebra.Graph.NonEmpty.AdjacencyMap".
+--
+-- @
+-- scc 'empty'      == 'empty'
+-- scc ('vertex' x) == 'vertex' (NonEmpty.'NonEmpty.vertex' x)
+-- @
+scc :: (Ord a) => AM.AdjacencyMap a -> AdjacencyMap (NonEmpty.AdjacencyMap a)
+scc = AAM . AM.scc
+
+-- | Compute the /topological sort/ of a graph.
+-- 
+-- @
+-- topSort (1)             == [1]
+-- topSort (1 * 2 + 3 * 1) == [3,1,2]
+-- @
+topSort :: (Ord a) => AdjacencyMap a -> [a]
+topSort (AAM am) = Typed.topSort (Typed.fromAdjacencyMap am)
+
+-- | Compute the /Cartesian product/ of graphs.
+-- Complexity: /O(s1 * s2)/ time, memory and size, where /s1/ and /s2/
+-- are the sizes of the given graphs.
+--
+-- @
+-- box (1 * 2) (3 * 4) == (1, 3) * (1, 4) + (2, 3) * (2, 4) 
+--                      + (1, 3) * (2, 3) + (1, 4) * (2, 4)
+-- box (1 + 2) (3 + 4) == (1, 3) + (1, 4) + (2, 3) + (2, 4)
+-- @
+box :: (Ord a, Ord b) => AdjacencyMap a -> AdjacencyMap b -> AdjacencyMap (a, b)
+box (AAM x) (AAM y) = AAM $ AM.overlay (fO $ xy y x) (fO $ yx x y)
```

Comment:
Why do you reimplement this function instead of simply calling to `AM.box`?

Reply:
Ah! Answering my own question: because there is no `AM.box` yet! I'll try to add it today, so that you can simplify this function.

Reply:
@adithyaov I've added `AM.box`, see #206.